src/todo_list.py

```diff
@@ -0,0 +1,198 @@
+from datetime import datetime, timedelta, time, date
+import pytz
+
+from enum_helper import NoValue
+from postgres import with_postgres_connection
+from json_record import JSONRecord
+
+
+class DaysOfTheWeek(str, NoValue):
+    MONDAY = 'Monday'
+    TUESDAY = 'Tuesday'
+    WEDNESDAY = 'Wednesday'
+    THURSDAY = 'Thursday'
+    FRIDAY = 'Friday'
+    SATURDAY = 'Saturday'
+    SUNDAY = 'Sunday'
+
+class MonthsOfTheYear(str, NoValue):
+    JANUARY = 'January'
+    FEBRUARY = 'February'
+    MARCH = 'March'
+    APRIL = 'April'
+    MAY = 'May'
+    JUNE = 'June'
+    JULY = 'July'
+    AUGUST = 'August'
+    SEPTEMBER = 'September'
+    OCTOBER = 'October'
+    NOVEMBER = 'November'
+    DECEMBER = 'December'
+
+class Goals2020(str, NoValue):
+    JOBS = "Get a job by August"
+    WEIGHT = "Get to 51kg by September"
+    BLOG = "Write at least one article a week in December"
+
+class Weekly(JSONRecord):
+    def __init__(self, number=None, days_of_the_week=None):
+        self.weekly_cadence = {
+            "number": number,
+            "days_of_the_week": days_of_the_week
+        }
+        super().__init__(self.weekly_cadence)
+    # todo
+
+class Monthly(JSONRecord):
+    def __init__(self, number=None, days_of_the_month=None):
+        self.monthly_cadence = {
+            "number": number,
+            "days_of_the_month": days_of_the_month
+        }
+        super().__init__(self.monthly_cadence)
+    # todo
+
+class Yearly(JSONRecord):
+    def __init__(self, number=None, dates=None):
+        self.yearly_cadence = {
+            "number": number,
+            "dates": dates
+        }
+        super().__init__(self.yearly_cadence)
+    # todo
+
+class ToDoList:
+    def __init__(self):
+        self.idx = None
+        self.task_name = None
+        self.created_on = None
+        self.start_date = None
+        self.end_date = None
+        self.due_date = None
+        self.once = False
+        self.yearly = Yearly
+        self.monthly = Monthly
+        self.weekly = Weekly()
+        self.daily = False
+        self.is_complete = {}
+        self.task_dependencies = {}
+        self.project = ""
+        self.subproject = ""
+        self.goal = Goals2020
+        self.habit = False
+        self.related_tasks = {} # this is for future use
+        # determine how long it takes to complete related projects/tasks on average
+        # to generate estimate for current project/task
+        # self.time_spent = {date: timedelta} # this tracks how long I've spent on this task(s)
+    def unpack_records(self, record):
+        # todo
+        # questions: how do you unpack everything
+        # questions: how do you enforce one element is one of those types
+        self.idx, self.task_name, self.created_on, self.end_date, self.due_date = record[:5]
+        print(self.task_name)
+        self.idx = record[0]
```

Comment:
For non-json store objects, i think psycopg2 has a way to bind a model to a query (or table? i don't remember) so that when it reads out the records it "unmarshals" the data directly into the object(s) for you

Reply:
i wasnt able to find this but i was able to do a workaround (not sure how amazing it is tho)